### ISSUE TITLE
Add option to function like `cargo init`

### ISSUE THREAD

**Issue by punkstarman** [cargo-generate/cargo-generate#193]
Cargo provides two commands for creating a new Rust project: `cargo new` and `cargo init`. The former creates a new subdirectory for the project and a SCM repo. The latter assumes the user has already created the project directory and maybe even the SCM repo and only adds the rest.

`cargo generate` works like `cargo new`, but sometimes the user has already created a Git repo and cloned it so they don't want to have to jump through extra hoops to get everything squared away.

```shell
git clone git@...:myproject.git
cd myproject.git
cargo generate --git git@...:mytemplate.git
ls
# :scream: Sacrébleu, mise en abyme! (and not Inception as most people say but that's an issue for a metaphysical repo)
mv myproject/* .
ls
ls -a
# :dizzy_face: why doesn't * pick up dot files?
mv myproject/.* .
ls -a
git status
# :skull: I overwrote the local repo
```

**Comment by punkstarman** [comment]
Thank you. I'll get to it.

**Comment by ashleygwilliams** [comment]
@punkstarman awesome. i am curious- what do you expect the end user command API to be? `cargo generate` is how we invoke today- what were you thinking for this feature?

**Comment by punkstarman** [comment]
@ashleygwilliams thanks. Good question, I was just thinking about it cause I want to write the test case.

I'm leaning toward adding a flag. I'm not sure of what name yet.
- `--init`
- `--bare`
- `--no-create-dir`
- :man_shrugging: 

I'm shopping around for other options. This CLI doesn't use subcommands so I'm reluctant to go there.

What do you think about this?

**Comment by avrong** [comment]
@ashleygwilliams @punkstarman Any updates on this? Wondering if there anything I can help you with.

**Comment by punkstarman** [comment]
@avrong, thanks for the offer.

My main problem is debugging the build failure on Windows. I haven't had a computer with Windows installed for a few years now, and I am very much allergic to that OS.

But if you could help out with that, I would be very grateful!

**Comment by k0pernicus** [comment]
@punkstarman Is it a permission issue? Like you are trying to overwrite a directory without the permissions for that...

**Comment by k0pernicus** [comment]
```sh
---- no_create_dir::it_uses_existing_directory_when_passing_no_create_dir_flag stdout ----
thread 'no_create_dir::it_uses_existing_directory_when_passing_no_create_dir_flag' panicked at 'Unexpected failure.
code-1
stderr=```Error:  Git Error: failed to remove directory `C:\projects\cargo-generate\target\generated-tests\test-25-main`
```
command=`"C:\\projects\\cargo-generate\\target\\debug\\cargo-generate.exe" "generate" "--git" "C:\\projects\\cargo-generate\\target\\generated-tests\\test-25-template" "--name" "foobar-project" "--no-create-dir"`
code=1
stdout=``````
stderr=```Error:  Git Error: failed to remove directory `C:\projects\cargo-generate\target\generated-tests\test-25-main`
```
', C:\Users\appveyor\.cargo\registry\src\github.com-1ecc6299db9ec823\assert_cmd-0.4.0\src\assert.rs:140:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrac
```
Checking at logs, it seems a permission issue actually...
I have a windows machine - I can help on this @punkstarman if you want

**Comment by punkstarman** [comment]
@k0pernicus, that would be great.

Why is Windows encountering a permission error and not Linux or macOS?

**Comment by k0pernicus** [comment]
@punkstarman sorry, I made a mistake understanding your integration test...  
To be honest, I don't have any clue of what is happening for Windows build 🤔 
Will take a look asap more closely

**Comment by k0pernicus** [comment]
Otherwise, I don't know why, but it seems that 

```sh
C:\\projects\\cargo-generate\\target\\debug\\cargo-generate.exe generate --git C:\\projects\\cargo-generate\\target\\generated-tests\\test-25-template --name foobar-project --no-create-dir
```

is deleting the folder instead of cloning the content in...

**Comment by k0pernicus** [comment]
So, I took a look on my Windows machine and it seems an issue with 

```rust
db.copy_to(rev, project_dir.as_path(), &config)?;
```

line 54 in src/git.rs.

Indeed, it seems that the `clone_to`, on Windows, try to remove the current folder to clone the changes in...

I don't really know if this is configuration option that we miss, if there is a bug in git2, or if the behaviour of git implementation on Windows is totally different than *nix platforms (like GNU/Linux or macOS)...

You have different options here I would say:
1. Update `git2` and `cargo` dependencies to latest version to know if this is a bug from the lib... but as `cargo init` is using those libs I don't think so :/
2. Create another function than `create` (like `clone_into`) if `--no-create-dir` is provided by the user, that does not uses the `copy_to` function from `GitDatabase`.

**Comment by punkstarman** [comment]
@k0pernicus, thankyou so much for taking the time to make this analysis and narrow down the root problem.

Would you be willing to experiment with the second option? At first, anything that makes the test pass will do. We can always clean things up afterwards.

If I tried to do this relying only on the CI, my cycle times would be enormous.

**Comment by k0pernicus** [comment]
@punkstarman I can take a look, yep.
Unfortunately, my schedule's pretty busy, so I think I will check the options this week-end.

Do not hesitate to ping me next week if you don't have any update from me ;-)

**Comment by punkstarman** [comment]
OK. No problem.

Thank you for the help that you have already provided :smile:

**Comment by k0pernicus** [comment]
Hi @punkstarman 

Sorry for the delay: I have to reinstall Windows from scratch and make the machine setup again because of a crash...  
I will try to take a look at this this week-end, or next week.

Sorry again.

**Comment by taurr** [comment]
@k0pernicus, @punkstarman 

#293 Adds the possibility to disable git initialization.
#290 Ensures that cargo-generate won't initialize a repo inside another repo

Together, don't they solve this??

**Comment by punkstarman** [comment]
@taurr, no. With only those features, `cargo-generate` still creates a new directory for the project.

**Comment by taurr** [comment]
@sassman this seems to be yet another one closed by #194 and it company

**Comment by taurr** [comment]
@sassman please close this one - we added the flag in the last version I believe
